modin/core/storage_formats/pandas/query_compiler_caster.py

```diff
@@ -45,13 +45,13 @@ class QueryCompilerCasterCalculator:
     """
 
     def __init__(self):
-        self._caster_costing_map = {}
-        self._data_cls_map = {}
+        self._compiler_class_to_cost = {}
+        self._compiler_class_to_data_class = {}
         self._qc_list = []
         self._qc_cls_set = set()
         self._result_type = None
 
-    def add_query_compiler(self, query_compiler):
+    def add_query_compiler(self, query_compiler: BaseQueryCompiler):
```

Comment:
```suggestion
    def add_query_compiler(self, query_compiler):
```

It looks like `query_compiler` might not be an instance of `BaseQueryCompiler` here, so I would remove the type annotation.

Reply:
Oh right. I added that in response to Jonathan's comment, but it can either be an instance or a class.